crates/shared/src/bad_token/token_owner_finder/token_owner_list.rs

```diff
@@ -0,0 +1,51 @@
+use std::collections::HashMap;
+
+use super::TokenOwnerProposing;
+use anyhow::Result;
+use ethcontract::H160;
+
+type Token = H160;
+type Owner = H160;
+
+pub struct TokenOwnerList {
+    owners: HashMap<Token, Vec<Owner>>,
+}
+
+impl TokenOwnerList {
+    pub fn new(owners: HashMap<Token, Vec<Owner>>) -> Self {
+        Self { owners }
+    }
+}
+
+#[async_trait::async_trait]
+impl TokenOwnerProposing for TokenOwnerList {
+    async fn find_candidate_owners(&self, token: H160) -> Result<Vec<Owner>> {
+        Ok(self.owners.get(&token).cloned().unwrap_or_default())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn seasolver_finder_constructor_empty() {
```

Comment:
nit: The tests are still named after the seasolver.

Reply:
Ah yes. Fixed.